consensus/types/src/execution_block_header.rs

```diff
@@ -89,3 +90,62 @@ impl ExecutionBlockHeader {
         }
     }
 }
+
+#[derive(Debug, Clone, PartialEq, Eq, RlpEncodable)]
+#[rlp(trailing)]
+pub struct EncodableExecutionBlockHeader<'a> {
+    pub parent_hash: &'a [u8],
+    pub ommers_hash: &'a [u8],
+    pub beneficiary: &'a [u8],
+    pub state_root: &'a [u8],
+    pub transactions_root: &'a [u8],
+    pub receipts_root: &'a [u8],
+    pub logs_bloom: &'a [u8],
+    pub difficulty: u64,
+    pub number: u64,
+    pub gas_limit: u64,
+    pub gas_used: u64,
+    pub timestamp: u64,
+    pub extra_data: &'a [u8],
+    pub mix_hash: &'a [u8],
+    pub nonce: &'a [u8],
+    pub base_fee_per_gas: u64,
+    pub withdrawals_root: Option<&'a [u8]>,
+    pub blob_gas_used: Option<u64>,
+    pub excess_blob_gas: Option<u64>,
+    pub parent_beacon_block_root: Option<&'a [u8]>,
+}
+
+impl<'a> From<&'a ExecutionBlockHeader> for EncodableExecutionBlockHeader<'a> {
+    fn from(header: &'a ExecutionBlockHeader) -> Self {
+        let mut encodable = Self {
+            parent_hash: header.parent_hash.as_bytes(),
+            ommers_hash: header.ommers_hash.as_bytes(),
+            beneficiary: header.beneficiary.as_bytes(),
+            state_root: header.state_root.as_bytes(),
+            transactions_root: header.transactions_root.as_bytes(),
+            receipts_root: header.receipts_root.as_bytes(),
+            logs_bloom: header.logs_bloom.as_slice(),
+            difficulty: header.difficulty.as_u64(), // TODO this might panic
+            number: header.number.as_u64(),         // TODO this might panic
+            gas_limit: header.gas_limit.as_u64(),   // TODO this might panic
+            gas_used: header.gas_used.as_u64(),     // TODO this might panic
```

Comment:
Why are we using `u64`s in the RLP version of this type and having to deal with this conversion? Shouldn't the RLP version be using `U256`?

Reply:
Ah, it's because `Uint256` from `ethereum-types` doesn't implement `Rlp{Encodable,Decodable}`.

We are working on switching in this PR:

- https://github.com/sigp/lighthouse/pull/6078

If you don't mind @dospore, @eserilev can merge your PR into his? Or we could merge yours first, but it will need to convert between the two `U256` types